services/121-service/src/payments/payments.service.ts

```diff
@@ -420,46 +421,60 @@ export class PaymentsService {
   public async importFspReconciliationData(
     csvFile,
     programId: number,
+    payment: number,
+    fspIds: number[],
     userId: number,
   ): Promise<ImportResult> {
     const validatedImportRecords = await this.csvToValidatedFspReconciliation(
       csvFile,
     );
-
     let countImported = 0;
     let countNotFound = 0;
 
     const importResponseRecords = [];
     for await (const record of validatedImportRecords) {
       const importResponseRecord = record as ImportFspReconciliationResult;
 
-      // For now assume BoB Finance
-      // const registration = await this.bobFinanceService.findRegistrationFromInput(
-      //   record,
-      // );
-      // if (!registration) {
-      //   importResponseRecord.importStatus = ImportStatus.unmatched;
-      //   importResponseRecords.push(importResponseRecord);
-      //   countNotFound += 1;
-      //   continue;
-      // }
+      let registration, paTransactionResult;
+      // Loop over potentially multiple fsp's in same dataset
+      for (const fspId of fspIds) {
+        const fsp = await this.fspService.getFspById(fspId);
+
+        if (fsp.fsp === FspName.vodacash) {
+          console.log('fsp: ', fsp);
+          registration = null; //DUMMY > REMOVE
+          // const registration = await this.vodacashService.findRegistrationFromInput(
```

Comment:
@PeterSmallenbroek FYI: this is the part where I'm expecting our code to connect